lib/utils/property-order.js

```diff
@@ -1,39 +1,162 @@
+'use strict';
+
 var ember = require('./ember');
 
 module.exports = {
+  determinePropertyType,
   reportUnorderedProperties: reportUnorderedProperties
 };
 
-function reportUnorderedProperties(node, context, toPropertyInfo) {
+const NAMES = {
+  'service': 'service injection',
+  'property': 'property',
+  'single-line-function': 'single-line function',
+  'multi-line-function': 'multi-line function',
+  'observer': 'observer',
+  'model': '"model" hook',
+  'lifecycle-hook': 'lifecycle hook',
+  'actions': 'actions hash',
+  'method': 'method',
+  'unknown': 'unknown property type',
+  'attribute': 'attribute',
+  'relationship': 'relationship',
+};
+
+function determinePropertyType(node, parentType) {
+  if (ember.isInjectedServiceProp(node.value)) {
+    return 'service';
+  }
+
+  if (parentType === 'component') {
```

Comment:
mostly just for consistency with the `model` condition below and the fact that this will be easier to extend should we want to add a second condition that only applies to components